src/goto-instrument/contracts/contracts.cpp

```diff
@@ -174,6 +174,17 @@ void code_contractst::check_apply_loop_contracts(
     return invariant_copy;
   };
 
+  // Process "loop_entry" history variables
+
+  std::map<exprt, exprt> parameter2history;
+  goto_programt history;
+
+  // Find and replace "loop_entry" expression in the "invariant" variable
+  replace_old_parameter(invariant, parameter2history, loop_head->source_location, mode, history);
+
+  // Create 'loop_entry' history variables
+  insert_before_swap_and_advance(havoc_code, loop_head, history);
```

Comment:
May be you meant:
```suggestion
  insert_before_swap_and_advance(goto_function.body, loop_head, history);
```
because `loop_head` is an instruction iterator into that, not `havoc_code`?